pennylane_lightning/src/Makefile

```diff
@@ -0,0 +1,25 @@
+.PHONY: help clean test benchmark
```

Comment:
I felt it is easier to deal with all C++-related code inside `pennylane_lightning/src` directory. But of course, it is not essential.

Reply:
I always worry using Makefiles since we are a multiplatform library (and Makefiles + Windows can cause problems). Also, given CMake generates a makefile (if we do a run in a specific directory), I always worry about their inclusion. Though, if it makes life easier for debug/development, I think it is ok. Though, if another option exists, I'd prefer that.

Reply:
I just removed `Makefile` within this directory.